app/questionnaire/placeholder_transforms.py

```diff
@@ -292,3 +298,33 @@ def _create_hyperlink(href: str, link_text: str) -> str:
 
     def list_item_count(self, list_to_count: Optional[Sized]) -> int:
         return self._operations.evaluate_count(list_to_count)
+
+    def option_label_from_value(self, value: str, answer_id: str) -> str:
+
+        answers = self.schema.get_answers_by_answer_id(answer_id)
+        option_label: str = next(
+            (
+                options["label"]
+                for answer in answers
+                for options in answer["options"]
+                if value in options["value"]
+            ),
+            "",
+        )
+
+        if isinstance(option_label, str):
+            label = option_label
+
+        elif isinstance(option_label, dict):
+            parser = placeholder_parser.PlaceholderParser(
+                language=self.locale,
+                answer_store=AnswerStore(),
+                list_store=ListStore(),
+                metadata=ImmutableDict({}),
+                response_metadata={},
```

Comment:
This won't work because it is not given any of the actual stores. We shouldn't just be initializing the stores here. You will need to have the actual store we create, so placeholders have answer data etc to use when rendering. You want the `PlaceholderRenderer` which will use the parser.